File: 2- SEGUNDO SEMESTRE/Programación Orientada a Objeto/cuarta unidad/17 nov/productos.py
import json
from urllib import request


pagina = request.urlopen("http://localhost/productos/productos.php")
datos = pagina.read().decode("utf-8")
print(datos)

print("_"*100)
lista = json.loads(datos) #convertimos el string a una lista
for x in lista:
  print(f"{x['codigo']} {x['descripcion']:25} {x['precio']}")

File: 2- SEGUNDO SEMESTRE/Programación Orientada a Objeto/cuarta unidad/17 nov/productos.php
<?php
header('Content-Type: application/json');

$server="localhost";
$usuario="root";
$clave="";
$base="productos";
$conexion=mysqli_connect($server,$usuario,$clave,$base) or die("problemas") ;
mysqli_set_charset($conexion,'utf8'); 

$datos = mysqli_query($conexion, "SELECT codigo, descripcion, precio from articulos");
$resultado = mysqli_fetch_all($datos, MYSQLI_ASSOC);
echo json_encode($resultado);        
?>